contributed_definitions/nyaml/NXsingle_crystal.yaml

```diff
@@ -0,0 +1,31 @@
+category: base
+doc: |
+  Description of a single crystal material or a single crystalline phase in a material.
+type: group
+NXsingle_crystal(NXobject):
+  sample_orientation(NX_FLOAT):

```

Comment:
Only for legacy support but for a general enough application of NeXus we have to be more creative my suggestion:
worked out already NXconventions, NXcoordinate_system, NXcoordinate_system_set

Reply:
Let's try to find compromises here maybe. Maybe the solution is a double structure with an easy-to-use interface for case of single crystals, and a complicated, but flexible one for more complicated structures? If this is well documented which to use when, this could work I think.